cli/src/commands/appUninstallAll.js

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable no-console */
+// @flow
+
+import { from } from "rxjs";
+import { mergeMap, filter, map, tap } from "rxjs/operators";
+import { withDevice } from "@ledgerhq/live-common/lib/hw/deviceAccess";
+import getDeviceInfo from "@ledgerhq/live-common/lib/hw/getDeviceInfo";
+import { initState, reducer, runAll } from "@ledgerhq/live-common/lib/apps";
+import { listApps, execWithTransport } from "@ledgerhq/live-common/lib/apps/hw";
+import { deviceOpt } from "../scan";
+
+export default {
+  description: "uninstall all apps in the device",
+  args: [deviceOpt],
+  job: ({ device }: $Shape<{ device: string }>) =>
+    withDevice(device || "")((t) => {
+      const exec = execWithTransport(t);
+      return from(getDeviceInfo(t)).pipe(
+        mergeMap((deviceInfo) =>
+          listApps(t, deviceInfo).pipe(
+            filter((e) => e.type === "result"),
+            map((e) => {
+              const { appByName, installed } = e.result;
+              // NB will uninstall those with dependecies first
+              const sortedInstalledApps = installed.sort(
+                (a, b) =>
+                  appByName[b.name].dependencies.length -
+                  appByName[a.name].dependencies.length
```

Comment:
it's good enough but i think a better would be to use app's logic called "wipe".

Reply:
Fair enough, didn't know it existed